distribution/lib/Standard/Base/0.0.0-dev/src/System/File_Format.enso

```diff
@@ -35,6 +40,17 @@ type Auto_Detect
                     @Tail_Call reader (idx + 1)
         reader 0
 
+    ## PRIVATE
+    get_web_parser : Text -> URI -> Any | Nothing
+    get_web_parser content_type uri =
+        reader idx =
+            if idx >= format_types.length then Nothing else
+                format = format_types.at idx . for_web content_type uri
+                if format.is_nothing.not then format else
+                    @Tail_Call reader (idx + 1)
+        reader 0
```

Comment:
Probably not for this PR, but I really don't like that we are getting a second instance of this loop in this file. One was bearable, two suggest we have a problem - IMO this should be doable using stdlib.

I assume the issue is that `map + find` will execute all descriptors whereas we want just the first one. I guess one solution would be to do some kind of lazy map variant?

We could do
```
## ADVANCED
Vector.lazy_map self f = Array_Proxy.new self.length (ix -> f (self.at ix))
```

or maybe figure something else. I wonder if we can encapsulate this logic nicely. A `map_and_find` function looks weirdly specific.

If we do not decide to add these tools to the stdlib, maybe we can at least encapsulate this `map_and_find` logic locally in this module? So that the two places where we use it use the same implementation.

I know it is a bit of a nitpick, as it is _just a loop_ and in many languages it is a totally normal thing to do. I guess my point is that in Enso such loops usually suggest something complicated is going on and the function here is not that easy to read in comparison to our usual code. So IMO splitting it would make the code much nicer. But it's not a very high priority thing I guess.

Reply:
Merged to a single implementation but if you want a different approach will need to be another PR.

Reply:
I think looking at also other usages, our stdlib needs a `Vector.lazy_map` operation. Probably with an `ADVANCED` tag as due to its performance implications it needs to be used carefully.

Definitely not something to do in this PR. Do you think such a method can be useful? If so, I can create a ticket for us to create it at some point.

I like adding such a method especially because then we give this pattern a _name_ and it becomes easier to find places in our code where it is used.